### PR TITLE
Demo

### PR DESCRIPTION
**Pull request by tejal29**
Demo feedback

### CODE REVIEW

File: src/python/main.py [hunk 1]
```diff
@@ -1,25 +1,51 @@
...
+                            search_term=options.search,
+                            colorize=options.colorize,
+                            force=options.force,
+                            base_url=options.base_url,
+                            ttl=ttl,
+                            access_token=options.access_token)
+  try:
+    if options.search:
+      grepg_client.search_user_topic_cheats()
+    else:
```
Comment (tejal29) [inline src/python/main.py]: @evidanary said add list_all_cheats.

File: src/python/main.py [hunk 2]
```diff
@@ -1,25 +1,51 @@
...
+from __future__ import print_function
 
+from optparse import SUPPRESS_HELP
+
+from twitter.common import app, log
+from twitter.common.log.options import LogOptions
+from lib.grepg_client import CommandError, GrepgClient
+from util.utils import get_config, get_user_name, print_util
+
+TTL = 86400  # 1 day
```
Comment (tejal29) [inline src/python/main.py]: default as 10 seconds

File: src/python/lib/grepg_client.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      print_util(cheat.description, 'blue', self.colorize)
+      print(cheat.command)
+
+  def search_user_topic_cheats(self):
+    (since_timestamp, user_topic_cheats) = self._get_user_topic_cheats()
+    print_util('User: {0}, Topic: {1}, Search-Term: {2}, Last fetched: {3} ago'
+       .format(self.user_name, self.topic, self.search_term, since_time_in_words(since_timestamp)),
+      'green', self.colorize)
+
+    search_results = filter(lambda topic_cheat: self.search_term in topic_cheat.description, user_topic_cheats)
```
Comment (tejal29) [inline src/python/lib/grepg_client.py]: do "and" and "or"

File: README.md [hunk 4]
```diff
@@ -1 +1,141 @@
-# grepg
+grepg
+===
+
+`grepg` (pronounced Grep G) is a python client for [GrepPage](https://www.greppage.com).  It allows you to access your cheat sheets without leaving the terminal.
+
+![grepg screenshot](https://github.com/tejal29/grepg/raw/master/img/screenshot.png)
```
Comment (evidanary) [inline README.md]: currently broken.

File: README.md [hunk 5]
```diff
@@ -1 +1,141 @@
...
+Now, you can do
+
+```
+$ grepg -t python
+User: tejal29, Topic: python, Last fetched: 5 second(s) ago
+...
+```
+
+#Advanced usage
+`grepg -s` does a simply does an 'or' when multiple search terms specified on command line and cheat descriptions. In order to get finer results, you can change the `match_op` to 'and'
```
Comment (evidanary) [inline README.md]: duplicate "does a"

File: README.md [hunk 6]
```diff
@@ -1 +1,141 @@
...
+c match {
+   case ‘,’ | ‘.’ => println(‘punctuation’)
+   case _       => println(‘not punctuation’)
+}
+
+```
+
+
+#Configuration
+Setup defaults in `~/.grepg.yml`
```
Comment (evidanary) [inline README.md]: " will fetch all the cheats from until the cache is expired" should be "is the ttl for the cache."

File: README.md [hunk 7]
```diff
@@ -1 +1,141 @@
...
+
+Options:
+  -h, --help, --short-help
+                        show this help message and exit.
+  --long-help           show options from all registered modules, not just the
+                        __main__ module.
+  -v, --verbose         Log to stdout [default: False]
+  -f, --force           Force fetch cheats and update cached cheats. Default
+                        Cache TTL 10 seconds [default: False]
+  -u USER, --user=USER  username
```
Comment (evidanary) [inline README.md]: is there a way to make this lowercase ? like - "user"

**Pull request closed without merging**